File: src/main/java/com/jdaniel/sakuappapi/task/model/TaskState.java
package com.jdaniel.sakuappapi.task.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "task_states", schema = "task")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class TaskState {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Short taskId;
    private String state;

    @OneToMany(mappedBy = "taskState", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Task> tasks;
}

File: src/main/java/com/jdaniel/sakuappapi/task/model/TaskState.kt
package com.jdaniel.sakuappapi.task.model

import jakarta.persistence.*
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

@Entity
@Table(name = "task_states", schema = "task")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
class TaskState {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var taskStateId: Short? = null
    var state: String? = null

    @OneToMany(mappedBy = "taskState", cascade = [CascadeType.ALL], orphanRemoval = true)
    var tasks: MutableList<Task?>? = null
}
